src/MessageBus/MessageBus.php

```diff
@@ -22,24 +22,22 @@ final class MessageBus implements IMessageBus
 	private $middleware = [];
 
 	/**
-	 * @var MiddlewareCallback|NULL
+	 * @var MiddlewareCallback
 	 */
-	private $cachedCallback = NULL;
+	private $cachedCallback;
 
 
 
-	public function __construct(?MiddlewareCallbackChainCreator $middlewareCallbackChainCreator = NULL)
+	/**
+	 * @param IMessageBusMiddleware[] $middleware
+	 * @param MiddlewareCallbackChainCreator|null $middlewareCallbackChainCreator
+	 */
+	public function __construct(array $middleware, ?MiddlewareCallbackChainCreator $middlewareCallbackChainCreator = NULL)
```

Comment:
Mabe consider use of variadic to add more type-safety `IMessageBusMiddleware ...$middleware`

Reply:
Will do.